backend/rpc-impl/src/main/java/org/sonarsource/sonarlint/core/rpc/impl/AnalysisRpcServiceDelegate.java

```diff
@@ -138,10 +142,31 @@ public void didSetUserAnalysisProperties(DidChangeAnalysisPropertiesParams param
     notify(() -> getBean(AnalysisService.class).setUserAnalysisProperties(params.getConfigurationScopeId(), params.getProperties()));
   }
 
+  @Override
   public void didChangeAutomaticAnalysisSetting(DidChangeAutomaticAnalysisSettingParams params) {
     notify(() -> getBean(AnalysisService.class).didChangeAutomaticAnalysisSetting(params.isEnabled()));
   }
 
+  @Override
+  public void analyzeFullProject(AnalyzeFullProjectParams params) {
+    notify(() -> getBean(AnalysisService.class).analyzeFullProject(params.getConfigScopeId(), params.isHotspotsOnly()));
+  }
+
+  @Override
+  public void analyzeFileList(AnalyzeFileListParams params) {
```

Comment:
Nitpick/personal taste: I prefer not to use a type in the method name, like here "list". I would suggest `analyzeFiles`

Reply:
The problem is we already have the `analyzeFiles` method. And I don't have a better name. We can discuss this before merging to master.